lib/galaxy/datatypes/data.py

```diff
@@ -369,7 +369,7 @@ def to_archive(self, dataset, name=""):
         """
         rel_paths = []
         file_paths = []
-        if dataset.datatype.composite_type or dataset.extension == 'html':
+        if dataset.datatype.composite_type or dataset.extension == '.html' or dataset.extension.endswith('.html'):
```

Comment:
```suggestion
        if dataset.datatype.composite_type or dataset.extension.endswith('html'):
```
Just simplified the check a bit here since `endswith` will cover the previous check and also I've removed the preceding `.` to make sure the old check is still covered. Hope this makes sense.